static/modules/entity.js

```diff
@@ -114,18 +128,76 @@ export class Entity {
     this.vel = vel;
     /** @type {Vector} */
     this.acc = acc;
+    this.collisionType = "Circle";
+  }
+
+  /**
+   * Set the collisionShape of the entity. Give no arguments to reset the shape
+   * to be calculated each call of getCollisonShape
+   * @param {CollisionShape | undefined} collisionShape
+   * @returns {void}
+   */
+  setCollisionShape(collisionShape) {
+    this.collisionShape = collisionShape;
+  }
+  /**
+   * Set the collisionShape of the entity when colliding with terrain. Give no
+   * arguments to reset the shape to be the same as collisionShape
+   * @returns {void}
+   */
+  setTerrainCollisionShape(terrainCollisionShape) {
+    this.terrainCollisionShape = terrainCollisionShape;
+  }
+
+  /**
+   * Returns the CollisionShape of the entity.
+   * If the entity has a collisionEntity property, that shape is updated to the
+   * current position and is returned. Otherwise, a new CollisionShape is
+   * calculated.
+   * @returns {CollisionShape}
+   */
+  getCollisionShape() {
+    if (this.collisionShape !== undefined) {
+      this.collisionShape.pos = this.pos;
+      this.collisionShape.vel = this.vel;
+      return this.collisionShape;
+    } else if (this.collisionType == "Box") {
+      return new Box(this.width, this.height, this.pos, this.vel);
+    } else if (this.collisionType == "Circle") {
+      return new Circle(
+        Math.min(this.width, this.height) / 2,
+        this.pos,
+        this.vel
+      );
+    }
+    return new CollisionShape("undefined", this.pos, this.vel);
+  }
+
+  /**
+   * Returns the terrainCollisionShape of the entity.
+   * If the entity has a collisionEntity property, that shape is updated to the
+   * current position and returned. Otherwise, the collisionShape is returned.
+   * @returns {CollisionShape}
+   */
+  getTerrainCollisionShape() {
+    if (this.terrainCollisionShape !== undefined) {
+      this.terrainCollisionShape.pos = this.pos;
+      this.terrainCollisionShape.vel = this.vel;
+      return this.terrainCollisionShape;
+    }
+    return this.getCollisionShape();
   }
 
   onScreen() {
     const { width: screenWidth, height: screenHeight } = getScreenDimensions();
-    const screenEntity = new Entity(
+    const screenBox = new Box(
```

Comment:
Sweet, it always felt a little wrong to have to create a dummy screen entity in order to check for collision.